### PR TITLE
Navigation updates

### PR DESCRIPTION
**Pull request by subsymbolic**
Reviewer: Caine
Asana: https://app.asana.com/0/361428290920652/384289442884654

**Description**:
• Adds the fire button to the browsing screen
• Renames the "Bookmark" option to "Add to Bookmarks"
• Adds a "New Tab" option to 3-dot menu
• Adds a new toolbar to the home and browsing screens to allow different button sets
• Moves "Settings" button
• Adds bookmark button to home omni bar
• Adds state machine to manage various state/button combinations for the omnibar
• Removes "Copy" option from 3-dot menu

**Steps to test this PR**:
Run app and check that the new navigation structure.

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR DRI Checklist:
- [x] Get advice or leverage existing code
- [x] Agree on technical approach with reviewer (if the changes are nuanced)
- [x] Ensure that there is a testing strategy (and documented non-automated tests)
- [x] Ensure there is a documented monitoring strategy (if necessary)
- [x] Consider systems implications (Database connections, Grafana stats, CPU)

### CODE REVIEW

File: Core/WKWebViewExtension.swift [hunk 1]
```diff
@@ -67,6 +67,7 @@ extension WKWebView {
         dataStore.fetchDataRecords(ofTypes: allData) { records in
             let externalRecords = records.filter { $0.displayName != Constants.cacheKey }
```
Comment (nilnilnil) [inline Core/WKWebViewExtension.swift]: This is what I was saying could probably become a method of some sort to take in a non const.

File: DuckDuckGo/Localizable.strings [hunk 2]
```diff
@@ -34,15 +34,17 @@
 "action.title.pasteAndGo" = "Paste & Go";
 "action.title.save" = "Save";
 "action.title.cancel" = "Cancel";
-"action.title.newTab" = "Open in New Tab";
+"action.title.newTab" = "New Tab";
```
Comment (nilnilnil) [inline DuckDuckGo/Localizable.strings]: Why is this file not enough to translate?

File: DuckDuckGo/OmniBar.swift [hunk 3]
```diff
@@ -171,37 +183,48 @@ class OmniBar: UIView {
...
         omniDelegate?.onContenBlockerPressed()
     }
 }
 
 extension OmniBar: UITextFieldDelegate {
     
     func textFieldDidBeginEditing(_ textField: UITextField) {
-        onEditingMode()
+        refreshState(state.onEditingStartedState)
+        DispatchQueue.main.async {
```
Comment (nilnilnil) [inline DuckDuckGo/OmniBar.swift]: Why is this construct necessary?

File: DuckDuckGo/OmniBarState.swift [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+
+import Foundation
+import Core
+
+protocol OmniBarState {
```
Comment (nilnilnil) [inline DuckDuckGo/OmniBarState.swift]: nice

File: DuckDuckGo/TabSwitcherViewController.swift [hunk 5]
```diff
@@ -114,12 +111,12 @@ class TabSwitcherViewController: UIViewController {
         delegate.tabSwitcher(self, didRemoveTabAt: index)
         collectionView.reloadData()
         refreshTitle()
-    }
-    
-    fileprivate func dismiss() {
         if tabsModel.isEmpty {
             delegate.tabSwitcherDidRequestClearAll(tabSwitcher: self)
         }
+    }
```
Comment (nilnilnil) [inline DuckDuckGo/TabSwitcherViewController.swift]: This was a confusing part of the diff to read at first; but I think I get it now. You want this clear all called on onClear, and dismiss just does dismiss.

File: DuckDuckGo/TabSwitcherViewController.swift [hunk 6]
```diff
@@ -114,12 +111,12 @@ class TabSwitcherViewController: UIViewController {
...
+    
+    fileprivate func dismiss() {
```
Comment (nilnilnil) [inline DuckDuckGo/TabSwitcherViewController.swift]: Why this? Seems like a loss in explicitness. Just for terseness?

File: Core/WKWebViewExtension.swift (hunk 1 above)
Comment (subsymbolic) [inline Core/WKWebViewExtension.swift]: Definitely. We'd pass in a array of items to blow away based on top level domains we hit during a page load. All webkit exposes for matching is this `displayName` so it probably won't be a perfect match, but we'll hopefully catch most items added to the cache.

File: DuckDuckGo/Localizable.strings (hunk 2 above)
Comment (subsymbolic) [inline DuckDuckGo/Localizable.strings]: The strings in Interface Builder use a different system and aren't yet extracted.

File: DuckDuckGo/OmniBar.swift (hunk 3 above)
Comment (subsymbolic) [inline DuckDuckGo/OmniBar.swift]: Do you mean the async? We're in the middle of starting editing and the ui stutters if we block to highlight.

File: DuckDuckGo/TabSwitcherViewController.swift (hunk 6 above)
Comment (subsymbolic) [inline DuckDuckGo/TabSwitcherViewController.swift]: Looks like you answered this in your other comment... Let me know if you want further clarification.

**Pull request merged**